cmake/modules/AddSwift.cmake

```diff
@@ -1805,34 +1826,56 @@ function(add_swift_library name)
             foreach(DEP ${SWIFTLIB_LINK_LIBRARIES})
               if (NOT "${DEP}" STREQUAL "icucore")
                 add_dependencies("${VARIANT_NAME}-static"
-                  "${DEP}-${SWIFT_SDK_${sdk}_LIB_SUBDIR}-static")
+                  "${DEP}-${dependency_suffix}-static")
               endif()
             endforeach()
           endif()
 
           # Note this thin library.
           list(APPEND THIN_INPUT_TARGETS ${VARIANT_NAME})
+
+          # For non-Darwin SDKs, we install each of the architectures
+          if(NOT ${sdk} IN_LIST SWIFT_APPLE_PLATFORMS)
+            list(APPEND "${resource_list}" "${VARIANT_NAME}")
+            set(install_libpath
+                "${SWIFTLIB_DIR}/${SWIFT_SDK_${sdk}_LIB_SUBDIR}/${arch}")
+            set_property(TARGET ${VARIANT_NAME} PROPERTY SWIFTLIB_INSTALL_FILES $<TARGET_FILE:${VARIANT_NAME}>)
+            set_property(TARGET ${VARIANT_NAME} PROPERTY SWIFTLIB_INSTALL_DESTINATION "${SWIFT_SDK_${sdk}_LIB_SUBDIR}/${arch}")
+            set_property(TARGET ${VARIANT_NAME} PROPERTY SWIFTLIB_INSTALL_EXPORT_TARGETS "${VARIANT_NAME}")
```

Comment:
Instead of these three, can you collapse them and use `set_target_properties` instead?

```
set_target_properties(${VARIANT_NAME}
                      PROPERTIES
                        SWIFTLIB_INSTALL_FILES $<TARGET_FILE:${VARIANT_NAME}>
                        SWIFTLIB_INSTALL_DESTINATION ${SWIFT_SDK_${sdk}_LIB_SUBDIR}/${arch}
                        SWIFTLIB_INSTALL_EXPORT_TARGETS ${VARIANT_NAME})
```